### PR TITLE
Python 3.12 migration

### PR DESCRIPTION
**Pull request by pavelzw**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: .ci_support/migrations/python312.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+python:
+  - 3.12.* *_cpython
+# additional entries to add for zip_keys
+numpy:
+  - 1.26
+python_impl:
+  - cpython
+channel_sources:
+  - py312-migration,conda-forge
```
Comment (h-vetinari) [inline .ci_support/migrations/python312.yaml:43]: This is going to need to point to the right label
```suggestion
  - conda-forge/label/python_rc,conda-forge
```

**Comment by pavelzw** [comment]
Depends on https://github.com/conda-forge/cross-python-feedstock/pull/74 and https://github.com/conda-forge/python_abi-feedstock/pull/22

**Comment by pavelzw** [comment]
You are right, this one was a bit too eager 😅

**Comment by pavelzw** [comment]
@conda-forge-admin, please rerender

File: .ci_support/migrations/python312.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            - 3.9.* *_cpython
+            - 3.10.* *_cpython
+            - 3.11.* *_cpython
+            - 3.12.* *_cpython  # new entry
+            - 3.6.* *_73_pypy
+            - 3.7.* *_73_pypy
+            - 3.8.* *_73_pypy
+            - 3.9.* *_73_pypy
+    paused: true
+    use_local: True
```
[the remark this answers is not in the document]
Comment (h-vetinari) [reply .ci_support/migrations/python312.yaml:20]: We should remove this, now that the migration is live.

**Comment by pavelzw** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
We're going to have to fix (or deactivate) the `run_test.py`. I guess in the interest of speed, we can deactivate it for now.

Do you mind if I force-push into this PR? I try to keep the commit-churn low for having a sensible history. The commit would still be yours of course.

**Comment by pavelzw** [comment]
> Do you mind if I force-push into this PR? I

feel free, I’m only on my phone right now

**Comment by h-vetinari** [comment]
I ended up opening a separate PR, because I had a more minimal fix for the test failures in mine. It looks like it's passing, so I'll probably merge #124 once it's green, which will supersede this PR.

**Pull request closed without merging**